File: Bai2/Employee.cpp
#include "Employee.h"
#include <string.h>
using namespace std;
Employee::Employee()
{
    this->firstname = "default";
    this->lastname = "default";
    this->birthdate = Date();
    this->hiredate = Date();
}
Employee::Employee(string fn ,string ln , Date bd , Date hd)
{
    this->firstname = fn;
    this->lastname = ln;
    this->birthdate = bd;
    this->hiredate = hd;
}

Date Employee::getBd()
{
    return birthdate;
}

Date Employee::getHd()
{
    return hiredate;
}

void Employee::print()
{
    cout<<"birth Date :"<<this->birthdate.print()<<endl;
    cout<<"Name:"<<this->firstname<<" " <<this->lastname<<endl;
    cout<<"Hire Date:" <<this->hiredate.print();
}

Employee::~Employee()
{

}

File: Bai2_python/employee.py
from date import Date

class Employee:
    def __init__(self, firstname="default", lastname="default", birthdate=Date(), hiredate=Date()):
        self.firstname = firstname
        self.lastname = lastname
        self.birthdate = birthdate
        self.hiredate = hiredate

    def get_bd(self):
        return self.birthdate

    def get_hd(self):
        return self.hiredate

    def print(self):
        print(f"Birth Date: {self.birthdate}")
        print(f"Name: {self.firstname} {self.lastname}")
        print(f"Hire Date: {self.hiredate}")
